onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -38,13 +38,13 @@ def get_shape_from_type_proto(type_proto):
     if type_proto.tensor_type.HasField("shape"):
         return [get_dim_from_proto(d) for d in type_proto.tensor_type.shape.dim]
     else:
-        return None  # note no shape is different from shape without dim (scalar)
+        return []  # note no shape is different from shape without dim (scalar)
 
 
 def get_shape_from_value_info(vi):
     cls_type = vi.type.WhichOneof("value")
     if cls_type is None:
-        return None
+        return []
```

Comment:
Line 52 returns None. Maybe change that as well.

A better solution is to for each reference of get_shape_from_value_info and get_shape_from_type_proto, we need to check the return value has shape value, then process accordingly.  For example,
```
    scan_input_dim = get_shape_from_type_proto(self.known_vi_[node.input[-1]].type)[scan_input_axes[-1]]
    if scan_input_dim:
        ...